### PR TITLE
docs: document type field in ipfs.get output

### PR DESCRIPTION
**Pull request by achingbrain**
Also adds a test for deeply nested directories

### CODE REVIEW

File: docs/core-api/FILES.md [hunk 1]
```diff
@@ -475,20 +476,24 @@ Here, each `path` corresponds to the name of a file, and `content` is an async i
...
     content.push(chunk)
   }
 
   console.log(content)
 }
 ```
 
+When invoking this method via the HTTP API client, the response arrives as a stream in the tar format containing the entire contents of the file or directory tree.
+
+If you are iterating over a directory, in order to proceeed to the next entry in the tar, you must consume the `content` field of the current entry if it is present.
```
Comment (hugomrdias) [inline docs/core-api/FILES.md:495]:
```suggestion
If you are iterating over a directory, in order to proceed to the next entry in the tar, you must consume the `content` field of the current entry if it is present.
```

File: docs/core-api/FILES.md [hunk 2]
```diff
@@ -475,20 +476,24 @@ Here, each `path` corresponds to the name of a file, and `content` is an async i
...
-for await (const chunk of file.content) {
+  for await (const chunk of file.content) {
     content.push(chunk)
   }
 
   console.log(content)
 }
 ```
 
+When invoking this method via the HTTP API client, the response arrives as a stream in the tar format containing the entire contents of the file or directory tree.
```
Comment (hugomrdias) [inline docs/core-api/FILES.md:493]: the user doesnt care about stream being a tar, maybe its best to not mention tar here to avoid confusion

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: docs/core-api/FILES.md (hunk 2 above)
Comment (achingbrain) [inline docs/core-api/FILES.md:493]:
```suggestion
When invoking this method via the HTTP API client, the response arrives as a stream containing either the entire contents of the file (if the passed [CID][] resolves to a file) or recursive directory tree and all files contained therein (if the passed [CID][] resolves to a directory).
```

Comment (achingbrain) [reply docs/core-api/FILES.md]: Good shout, I've updated the text

File: docs/core-api/FILES.md [hunk 3]
```diff
@@ -475,20 +476,24 @@ Here, each `path` corresponds to the name of a file, and `content` is an async i
...
     content.push(chunk)
   }
 
   console.log(content)
 }
 ```
 
+When invoking this method via the HTTP API client, the response arrives as a stream containing either the entire contents of the file (if the passed [CID][] resolves to a file) or recursive directory tree and all files contained therein (if the passed [CID][] resolves to a directory).
+
+If you are iterating over a directory, in order to proceed to the next entry in the tar, you must consume the `content` field of the current entry if it is present.
```
Comment (hugomrdias) [inline docs/core-api/FILES.md:495]:
```suggestion
If you are iterating over a directory, in order to proceed to the next entry in the stream, you must consume the `content` field of the current entry if it is present.
```

**Review by hugomrdias — CHANGES_REQUESTED** [review]

**Review by hugomrdias — APPROVED** [review]

**Pull request merged**